src/Storages/MergeTree/IMergeTreeDataPartWriter.cpp

```diff
@@ -90,7 +90,14 @@ ASTPtr IMergeTreeDataPartWriter::getCodecDescOrDefault(const String & column_nam
 {
     auto get_codec_or_default = [&](const auto & column_desc)
     {
-        return column_desc.codec ? column_desc.codec : default_codec->getFullCodecDesc();
+        if(column_desc.codec)
+            return column_desc.codec;
+
+        auto default_compression_codec_mergetree_settings = storage_settings->default_compression_codec.toString();
+        if(!default_compression_codec_mergetree_settings.empty())
+            return CompressionCodecFactory::instance().get(default_compression_codec_mergetree_settings)->getFullCodecDesc();
+
+        return default_codec->getFullCodecDesc();
```

Comment:
So a few things:
First of all, there are three very familiar functions in `StorageSnapshot`: https://github.com/ClickHouse/ClickHouse/blob/2315991504b1e95d7bb2594e54e3c6f749897d79/src/Storages/StorageSnapshot.cpp#L147C38-L186.
- `StorageSnapshot::getCodecDescOrDefault` is not used at all, it can be removed
- `StorageSnapshot::getCodecOrDefault` is only used in `LogSink` I think, so it can be moved into `LogSink`.

With these changes I would like to avoid the confusion of which function should be used when etc.


Second, this line `return default_codec->getFullCodecDesc();` shouldn't be executed at all. Therefore it makes sense to convert it to a `LOGICAL_ERROR`. So instead of returning, we should throw an exception with `LOGICAL_ERROR` with a message something about unexpected column name.

I don't think you have to do these changes, but wanted to document this somewhere public and this PR seemed the best place to do it.